# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1284425864 14400
#      Mon Sep 13 20:57:44 2010 -0400
# Node ID f5c863803e29b94dd4701002fa10ee4adf71692f
# Parent  2470b1bf191aefc07268b9f9fd76d1bc8e983744
gnuplot_drawnow.m: Use new function __gnuplot_has_terminal__().

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2010-09-13  Ben Abbott <bpabbott@mac.com>
 
+	* plot/gnuplot_drawnow.m: Use new function __gnuplot_has_terminal__().
 	* plot/private/__gnuplot_has_terminal__.m: New function.
 
 2010-09-13  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/common_size.m: Partially rewrite using cellindexmat.
 
 2010-09-13  Jaroslav Hajek  <highegg@gmail.com>
 
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -36,24 +36,17 @@ function gnuplot_drawnow (h, term, file,
   if (nargin >= 3 && nargin <= 5)
     ## Produce various output formats, or redirect gnuplot stream to a
     ## debug file.
     plot_stream = [];
     fid = [];
     default_plot_stream = get (h, "__plot_stream__");
     unwind_protect
       plot_stream = __gnuplot_open_stream__ (2, h);
-      if (__gnuplot_has_feature__ ("variable_GPVAL_TERMINALS"))
-        available_terminals = __gnuplot_get_var__ (plot_stream, "GPVAL_TERMINALS");
-        available_terminals = regexp (available_terminals, "\\b\\w+\\b", "match");
-        gnuplot_supports_term = any (strcmpi (available_terminals,
-                                              gnuplot_trim_term (term)));
-      else
-        gnuplot_supports_term = true;
-      endif
+      gnuplot_supports_term = __gnuplot_has_terminal__ (term, plot_stream);
       if (gnuplot_supports_term)
         enhanced = gnuplot_set_term (plot_stream (1), true, h, term, file);
         __go_draw_figure__ (h, plot_stream(1), enhanced, mono);
         if (nargin == 5)
           fid = fopen (debug_file, "wb");
           enhanced = gnuplot_set_term (fid, true, h, term, file);
           __go_draw_figure__ (h, fid, enhanced, mono);
         endif
